esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -411,10 +411,99 @@ static void ensureMergeOrderedEsdlTransform(Owned<IPropertyTree> &dest, IPropert
     mergePTree(dest, copy);
 }
 
+void EsdlServiceImpl::configureBindingAccessMap(const IPTree* node, const char* service, const char* method, const char* serviceSecurity)
+{
+    StringBuffer methodSecurity;
+    StringArray  accessList;
+    StringBuffer accessName;
+    StringBuffer accessLevel;
+    Owned<MethodAccessMap> methodAccess;
+
+    if (!isEmptyString(serviceSecurity))
+    {
+        // Service and method access requirements are combined by default, but the method may opt
+        // to ignore the service requirements.
+        //
+        // Consider a service level requirement of "{$method}Access" that applies a method-specific
+        // requirement to every method. Some methods, such as Ping or EchoTest, may require no
+        // security. The 'inherited' requirement may be incorrect for other methods, perhaps because
+        // the name is too long. Methods must be able to override the service's settings.
+        if (!node->getPropBool("@override_service_auth_feature", false))
+            methodSecurity.set(serviceSecurity).append(',');
+    }
+    methodSecurity.append(node->queryProp("@auth_feature"));
+    methodSecurity.replaceString("{$service}", service);
+    methodSecurity.replaceString("{$method}", method);
+
+    accessList.appendList(methodSecurity, ",");
+    if (accessList.ordinality() == 0)
+    {
+        // Either the binding's service and method both omit the auth_feature attribute or service
+        // level security is given and the method explicitly overrides it without specifying its
+        // own entries. The binding defers.
+        return;
+    }
+
+    // At least one security entry exists in the binding. The binding will define a method access
+    // map (even if empty) so it can take precedence.
+
+    methodAccess.setown(new MethodAccessMap());
+    m_bindingAccessMaps.setValue(method, methodAccess.getLink());
+    ForEachItemIn(idx, accessList)
+    {
+        auto accessEntry = accessList.item(idx);
+        auto entryLen = strlen(accessEntry);
+        decltype(entryLen) entryIdx = 0;
+        SecAccessFlags flag = SecAccess_Unknown;
+
+        accessName.clear();
+        for (; entryIdx <= entryLen && accessEntry[entryIdx] != ':'; entryIdx++)
+        {
+            if (!isspace(accessEntry[entryIdx]))
+                accessName.append(accessEntry[entryIdx]);
+        }
+        if (strieq(accessName, "None") || strieq(accessName, "Deferred"))
+        {
+            // No map entry is created for this list item, but other list items may still yield
+            // map entries. It is the binding author's responsibility to use "none" logically.
+            continue;
+        }
+        if (':' == accessEntry[entryIdx])
+        {
+            entryIdx++;
+            if (accessName.isEmpty())
+                accessName.setf("%sAccess", service);
+        }
+        else if (isEmptyString(accessName)) // an empty string will have length 1 for the NULL terminator
```

Comment:
Comment is still confusing.  Maybe because byte length and string length are different concepts.  I don't think the comment adds value.

Reply:
To reiterate my previous comment, it is an attempt to explain why I passed a StringBuffer, which has isEmpty, to isEmptyString when isEmpty was used three lines earlier. The value I see in the comment is that if I was reviewing the code I would object to the use of isEmptyString on a StringBuffer, and the comment suggests there was a reason for doing it.

Reply:
Again, I'd rather the weirdness be fixed, but if not the comment should clarify why the extra null is there, so the real reason can be looked at and understood in the future.